### PR TITLE
Various changes to downloadBuild.py

### PR DESCRIPTION
**Pull request by jschwartzentruber**
Not submitting the TaskCluster change yet as I need to test with bot.py & autobisect.

### CODE REVIEW

File: util/downloadBuild.py [hunk 1]
```diff
@@ -387,9 +381,9 @@ def mozPlatform(arch):
         raise Exception("The arch passed to mozPlatform must be '64', '32', or None")
 
 
-def defaultBuildType(repoName, arch, debug):
+def defaultBuildType(repoName, arch, asan, debug):
```
Comment (nth10sd) [inline util/downloadBuild.py]: This change will break bot.py and autoBisect, as they use this function.

**Comment by nth10sd** [comment]
Thanks for cleaning up this code! I'd appreciate it if we could also unbreak bot.py and autoBisect as well, in this PR.

@jruderman you might want to look at the change for domInteresting.

**Comment by jschwartzentruber** [comment]
@nth10sd Shucks. I thought this was the safe part of my larger change to download from TaskCluster instead of Tinderbox (which definitely breaks bot.py etc). I'll get the whole thing working here. Thanks!

File: util/downloadBuild.py [hunk 2]
```diff
@@ -245,7 +246,7 @@ def downloadBuild(httpDir, targetDir, jsShell=False, wantSymbols=True, wantTests
                     localfile = os.path.join(buildDir, filename)
                     downloadURL(remoteURL, localfile, quiet=True)
                 gotApp = True
-            if remotefn.endswith('.mac.dmg') or remotefn.endswith('.mac64.dmg'):
+            elif re.search(r'(\.mac(64)?\.|^target)\.dmg$', fn):
```
Comment (jruderman) [inline util/downloadBuild.py]: This incorrectly expects two consecutive dots. I think there shouldn't be a `\.` between `(64)?` and `|`.

**Comment by nth10sd** [comment]
I know we are going to eliminate `tooltool` soon, but some of these changes look like they can land anyway, since `downloadBuild` is likely to be refactored to call `fuzzfetch` in the future instead of `tooltool`.

After we rebase this PR, the last rev d5a6e20 is no longer needed (it's in domfuzz which is now removed), so if we set `asan=False` by default in `defaultBuildType` and look at jruderman's comment, it should be ready to land.

@jschwartzentruber, what do you think?

**Pull request merged**